src/main/java/Duke.java

```diff
@@ -1,10 +1,294 @@
+import task.*;
+import storage.Storage;
+import ui.Ui;
+
+import java.util.*;
+
 public class Duke {
+
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+
+    public Duke() {
+
+        this.ui = new Ui();
+        this.storage = new Storage(System.getProperty("user.dir") + "/data/", "duke.txt");
+        this.taskList = storage.load();
+
+    }
+
+    public void run() {
+
+        Scanner sc = new Scanner(System.in);
+
+        ui.printWelcomeMessage();
+
+        boolean isExit = false;
+
+        while(sc.hasNextLine()) {
+
+            String input = sc.nextLine();
+
+            String[] tokens = input.split(" ", 2);
+
+            String command = tokens[0];
+            String argument = tokens.length == 2 ? tokens[1] : null;
+
+            if (command.equals("bye")) {
+                isExit = executeByeCommand();
+
+            } else if (command.equals("list")) {
+                isExit = executeListCommand();
+
+            } else if(command.equals("done")) {
+                isExit = executeDoneCommand(argument);
+
+            } else if (command.equals("todo")) {
+                isExit = executeTodoCommand(argument);
+
+            } else if (command.equals("deadline")) {
+                isExit = executeDeadlineCommand(argument);
+
+            } else if (command.equals("event")) {
+                isExit = executeEventCommand(argument);
+
+            } else if (command.equals("delete")) {
+                isExit = executeDeleteCommand(argument);
+
+            } else if (command.equals("find")) {
+                isExit = executeFindCommand(argument);
+
+            } else {
+                System.out.println("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+
+            if (isExit) {
+                break;
+            }
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        new Duke().run();
+
```

Comment:
Should there be an extra line here? Noticed it in a few places.

Reply:
Thank you for pointing it out!